### PR TITLE
Removing ids with hashEnabled when crypto.subtle is undefined

### PR DESCRIPTION
**Pull request by jileeshs**
https://jira.corp.adobe.com/browse/CORE-38489

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html) or I'm an Adobe employee.
- [ ] I have submitted a [documentation](https://github.com/AdobeDocs/alloy-docs) pull request or no changes are needed.
- [ ] I have made any necessary test changes and all tests pass.
- [ ] I have run the Sandbox successfully.

### CODE REVIEW

File: src/utils/convertStringToSha256Buffer.js [hunk 1]
```diff
@@ -10,6 +10,9 @@ const encodeText = str => {
 export default message => {
   const data = encodeText(message);
   const crypto = window.msCrypto || window.crypto;
+  if (!crypto.subtle) {
```
Comment (jfkhoury) [inline src/utils/convertStringToSha256Buffer.js]: I would log a warning here that we are not able to hash IDs on a non-secure page. Maybe we should double check that the protocol is HTTP before we log. So, inside `if (!crypto.subtle)..` add another if to check the protocol then warn the user.

Comment (jileeshs) [reply src/utils/convertStringToSha256Buffer.js]: Should we pass logger to the util?
What is the value in checking for HTTP, if the browser doesn't support hashing.
Shouldn't the warning be
`Your browser doesn't support hashing...`?

Comment (jfkhoury) [reply src/utils/convertStringToSha256Buffer.js]: As discussed on the call, let's not worry about the protocol.

**Comment by Aaronius** [comment]
I believe the decision was to fail the command (throw an error) if hashing is enabled on any ID and `crypto.subtle` is undefined, right?

**Comment by jfkhoury** [comment]
> I believe the decision was to fail the command (throw an error) if hashing is enabled on any ID and `crypto.subtle` is undefined, right?

Product wants to warn in the console that the IDs that need hashing have been dropped because hashing is not supported, but still make the `setCustomerIds` call with the remaining IDs, if any.

**Comment by liljenback** [comment]
@Aaronius , @jfkhoury , @jonsnyder , I think Jon suggested earlier that it would be nice to get 3rd party cookie info. This is another example it would be nice to get feedback on how often this fails. It would be nice if we could have a module like this in alloy that collects misc valuable stats and piggybacks that info on the next outbound call.

**Comment by Aaronius** [comment]
@jileeshs according to product requirements that @jfkhoury outlined in https://github.com/adobe/alloy/pull/310#issuecomment-566145063, we still need a warning message logged when an ID can't be hashed. Are you able to do this today? If not, let me know and I can help. I'd like to get it in for tomorrow's release.

**Pull request closed without merging**